### DISCUSSION TITLE
How to connect to sftpd automatically with DNS-SD feature?

### DISCUSSION THREAD

**Discussion by GHolk** [wolpi/prim-ftpd#302] [Q&A]
I'm wondering how to make use of the DNS-SD feature on linux.
Primitive FTPd will claim its name on lan, and user can change the service name, but the device name is not configurable (usually `Android.local`). If there are 2 android device in LAN, it could be Android.local or Android-2.local, so I don't know how to write my `ssh_config`.

I installed avahi on my linux, so I can resolve android.local to the ip address and write `Android.local` or `Android-2.local` in my ssh config, but how can I make use of DNS-SD so I can connect to ftpd with the service name?

I can write a script to parse the output of avahi-browse and find the ip/port with service name, and connect to it with ProxyCommand nc, but I don't think this is the right way. Can you share that how you use this feature?

```
Host pmf
ProxyCommand nc $(avahi-resolve-by-service-name my-prim-sftp)
HostKeyAlias my-prim-sftp
```

I also use another ssh daemon app [SSHelper](https://arachnoid.com/android/SSHelper/index.html).
SSHelper will use device's name as mDNS's name, like `Fairphone_4.local`, so it will less likely to occur collision.

Anywan, thank you for this useful app.

**Comment by wolpi** [comment]
I'm not using any of those auto discovery things. In my network usually all devices get same IP again and again.  I just use known IP.

**Comment by GHolk** [comment]
@abderrahim I see you add this feature. Would you like to provide some use cases?
